rust/src/lib.rs

```diff
@@ -712,6 +712,36 @@ pub extern "C" fn zcashlc_derive_extended_full_viewing_key(
     unwrap_exc_or_null(res)
 }
 
+/// Derives a Sapling address from the given Sapling extended full viewing key.
+///
+/// # Safety
+///
+/// - `extfvk` must be non-null and must point to a null-terminated UTF-8 string representing
+///   a Bech32-encoded Sapling extended full viewing key for the given network.
+/// - The memory referenced by `extfvk` must not be mutated for the duration of the function call.
+/// - Call [`zcashlc_string_free`] to free the memory associated with the returned pointer
+///   when done using it.
+#[no_mangle]
+pub extern "C" fn zcashlc_derive_sapling_address_from_viewing_key(
```

Comment:
Where or why is this needed? I don't think we should be exposing Sapling addresses after this SDK upgrade (the Sapling addresses are being merged into Unified Addresses), but if there's a use case for this I'd like to know.

Reply:
I think that this an intermediate step to get the current front-end wallet code working with current librustzcash.

Reply:
This is mostly for a validation done in the shield funds function on swift to make sure that the recipient of this transaction is "self". Ideally this will go entirely but being that the shield_funds function on rust crate (not the ffi) requires a extfvk we are forced to maintain this until we change that. 

Also. we can review if such validation is needed and decide whether ir should be done or not

Reply:
Oh, I see what you mean: we can't get an `extfvk` format out of a UFVK because we don't store everything needed. In that case, we can leave the `extfvk` argument to `zcashlc_shield_funds` and I'll fix that in a subsequent PR.

I still don't think that this method is necessary; the SDK is already trusting the Rust side to do everything else, so why wouldn't it trust the Rust side to correctly shield the funds to an address underneath `account`?

Reply:
I've removed this commit, pursuant to zcash/librustzcash#615 merging.